tensorflow/lite/core/subgraph.cc

```diff
@@ -740,6 +740,19 @@ TfLiteStatus Subgraph::Invoke() {
 TfLiteStatus Subgraph::ResizeTensor(TfLiteContext* context,
                                     TfLiteTensor* tensor,
                                     TfLiteIntArray* new_size) {
+  // If the dimensions don't change, avoiding
+  // unnecessary (re)allocations.
+  //
+  // Note that it's required to check `tensor->data.raw != nullptr`. Otherwise
+  // the subgraph won't allocate memory for a dynamic tensor when its size
+  // is equal to the original tensor size.
+  if (tensor->data.raw != nullptr &&
+      EqualArrayAndTfLiteIntArray(tensor->dims, new_size->size,
```

Comment:
Should this logic be in ResizeTensorImpl? The same logic is in ResizeInputTensor.

Reply:
> Should this logic be in ResizeTensorImpl? The same logic is in ResizeInputTensor.

@jdduke : In fact this thought came to me as well initially, but as there is a state change associated to this check in ResizeInputTensor(), i did not move this logic into ResizeTensorImpl(). Which i felt is correct as the state change should not be part of util implementation.

Please let me know, your valuable opinion, thanks!